src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp

```diff
@@ -137,26 +139,44 @@ bool connectionParameters::cellValue(int x, int y, const Yuni::String& value)
 
 static bool checkLoopFlow(const Antares::Matrix<>* direct_ntc,
                           const Antares::Matrix<>* indirect_ntc,
-                          double cellvalue,
-                          int row)
+                          double loopflow,
+                          int row,
+                          bool useLoopFlow)
 {
+    if (!useLoopFlow)
+        return true;
+
     for (int ts = 0; ts < direct_ntc->width; ts++)
     {
         const double ntcDirect = direct_ntc->entry[ts][row];
         const double ntcIndirect = indirect_ntc->entry[ts][row];
 
-        if (ntcDirect < cellvalue)
+        if (ntcDirect < loopflow)
         {
             return false;
         }
-        if (cellvalue < 0. && std::abs(cellvalue) > ntcIndirect)
+        if (loopflow < 0. && std::abs(loopflow) > ntcIndirect)
         {
             return false;
         }
     }
     return true;
 }
 
+static bool checkLoopFlowNTCDirect(double ntcDirect, double loopFlow, bool useLoopFlow)
```

Comment:
Function **checkLoopFlowNTCDirect** could be called inside function **checkLoopFlow**.
Same thing for **checkLoopFlowNTCIndirect**.